### PR TITLE
Add execute bit to postinstall script

### PR DESCRIPTION
**Pull request by ericstj**
Workaround https://github.com/dotnet/arcade/issues/7954

I regressed this when switching to use a template.  The template doesn't preserve the execute bit from the scripts.  That persists to the final package which then fails to install.

### CODE REVIEW

**Comment by ericstj** [comment]
Note to reviewers: this bug is blocking install of mac installers from main.  We've already merged into RC2.  Can I get a signoff here to unblock the 7.0 installers? https://github.com/dotnet/installer/issues/12094

**Review by sfoslund — APPROVED** [review]

**Pull request merged**
